packages/babel-plugin/src/utils/css-builders.tsx

```diff
@@ -0,0 +1,164 @@
+import * as t from '@babel/types';
+import kebabCase from '@compiled/ts-transform-css-in-js/dist/utils/kebab-case';
+import { addUnitIfNeeded } from '@compiled/ts-transform-css-in-js/dist/utils/css-property';
+import {
+  cssAfterInterpolation,
+  cssBeforeInterpolation,
+} from '@compiled/ts-transform-css-in-js/dist/utils/string-interpolations';
+import { hash } from '@compiled/ts-transform-css-in-js/dist/utils/hash';
+import generate from '@babel/generator';
+import { joinExpressions } from './ast-builders';
+import { State } from '../types';
+
+export interface CSSOutput {
+  css: string;
+  variables: {
+    name: string;
+    expression: t.Expression;
+  }[];
+}
+
+const getInterpolation = <TNode extends {}>(expression: TNode | undefined, state: State) => {
+  if (t.isIdentifier(expression) && state.declarations) {
+    const declaration = state.declarations[expression.name];
+    if (t.isVariableDeclaration(declaration) && declaration.kind === 'const') {
+      // We only want to pick out variable declarations that are constants
```

Comment:
might be an obvious answer - but why?

Reply:
lets add some more context around why we are doing this.